File: go/echo.go
package main

import (
	"fmt"
	"io"
	"net"
	"os"
)

func main() {
	service := ":9377"
	tcpAddr, err := net.ResolveTCPAddr("tcp4", service)
	checkErr(err)
	listener, err := net.ListenTCP("tcp", tcpAddr)
	checkErr(err)
	for {
		conn, err := listener.Accept()
		if err != nil {
			continue
		}
		go handleClient(conn)
	}
}

func handleClient(conn net.Conn) {
	defer conn.Close()
	io.Copy(conn, conn)
}

func checkErr(err error) {
	if err != nil {
		fmt.Fprintf(os.Stderr, "Fatal error: %s", err.Error())
	}
}

File: node/echo.js
var net = require('net');

var server = net.createServer(function(socket) {
    socket.write('Echo Server\r\n');
    socket.pipe(socket);

    socket.on('error', function(err) {
        console.log('error');
    });
});

server.listen(8089, '127.0.0.1');